content/docs/io/reading_writing_data.md

```diff
@@ -6,7 +6,156 @@ menu:
     parent: io
 ---
 
-This page has not been worked on yet. If you'd like to contribute visit the [doc-push]
-repo.
 
-[doc-push]: https://github.com/tokio-rs/doc-push
+
+# Non-blocking I/O
+
+In [the overview] we mentioned briefly that Tokio's I/O types implement
+non-blocking variants of `std::io::Read` and `std::io::Write` called
+[`AsyncRead`] and [`AsyncWrite`]. These are an integral part of Tokio's
+I/O story, and are important to understand when working with I/O code.
+
+> Note: in this section, we'll primarily talk about `AsyncRead`, but
+> `AsyncWrite` is pretty much exactly the same, just for writing data to
+> an I/O resource (like a TCP socket) instead of reading from it.
+
+So, let's take a look at [`AsyncRead`] and see what all the fuzz is
+about:
+
+```rust,no_run
+use std::io::Read;
+pub trait AsyncRead: Read {
```

Comment:
`AsyncRead` having `std::io::Read` as a super trait is a mistake IMO. Unfortunately, we can't deprecate it w/o a breaking change. This will change in 0.2.

What do you think about omitting `std::io::Read` here and only just bring it up when needed? I personally am not sure what the right thing to do here is. I'm leaning towards **sticking with what you have**, but i thought that I would bring up the thought.

Reply:
I'm fine w/ this if you like it. I just wanted to point out that the super trait is going away